plugins/woocommerce/includes/wc-product-functions.php

```diff
@@ -165,14 +165,10 @@ function wc_delete_product_transients( $post_id = 0 ) {
 
 			if ( ! isset( $scheduled[ $cache_key ] ) ) {
 				$queue                   = WC()->queue();
-				$existing                = $queue->search(
-					array(
-						'hook'     => 'wc_delete_related_product_transients_async',
-						'args'     => array( 'post_id' => $post_id ),
-						'status'   => 'pending',
-						'group'    => 'wc_delete_related_product_transients_group',
-						'per_page' => 1,
-					)
+				$existing                = $queue->get_next(
+					'wc_delete_related_product_transients_async',
+					array( 'post_id' => $post_id ),
+					'wc_delete_related_product_transients_group'
 				);
 				$scheduled[ $cache_key ] = ! empty( $existing );
```

Comment:
NIT: According to the `search` documentation:

https://github.com/woocommerce/woocommerce/blob/245252a5acf29ae4cfba7dda75851e65c01558bd/plugins/woocommerce/includes/interfaces/class-wc-queue-interface.php#L94-L102

The function can return `WC_DateTime` or `null`. What do you think if we are more type-safe since we need specifically check for null? 
 


```php
$scheduled[ $cache_key ] = $existing !== null;
```

What do you think?

Reply:
Good catch I agree https://github.com/woocommerce/woocommerce/pull/57630/commits/b57f9b0a97be1cf8753b0e6bc46bcb08dea09a00